kedro/io/core.py

```diff
@@ -23,6 +23,9 @@
 from kedro import KedroDeprecationWarning
 from kedro.utils import load_obj
 
+if TYPE_CHECKING:
```

Comment:
`os` is innocuous enough that maybe we don't need this conditional at all?

Reply:
@astrojuanlu Done!